### PR TITLE
Bugfix falsh_map coverity and flash_map test

### PR DESCRIPTION
**Pull request by nvlsianpu**
- fix for coverity issues (#5739, #5738, #6109).
- fix for test of get sectors API.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Comment by carlescufi** [comment]
There is something strange here. Could it be that there is a misconfiguration somewhere so that Coverity sees an empty array of `flash_drivers_map` and `flash_dev`? Coverity runs on some of our test cases so I would just like to check if one of the tests is misconfigured so that flash_map is being included in the build but with empty arrays?

**Comment by nvlsianpu** [comment]
FLASH device support might be enabled while non of driver implementation is enabled - think that is the issue.

**Comment by carlescufi** [comment]
@nvlsianpu is this something we could fix? i.e. change Kconfig so that it requires a flash driver instead of adding the `if()` statement

**Comment by nvlsianpu** [comment]
I can introduce something like FLASH_DRIVER_IS_ENABLED keyword, selected by each driver enable configuration.

**Comment by carlescufi** [comment]
@nvlsianpu if that is an option I think it would be a better fix

**Comment by nvlsianpu** [comment]
@carlescufi updated

**Comment by carlescufi** [comment]
@nashif as discussed yesterday

File: arch/Kconfig [hunk 1]
```diff
@@ -276,6 +276,12 @@ config FLASH_HAS_PAGE_LAYOUT
 	  This option is enabled when the SoC flash driver supports
 	  retrieving the layout of flash memory pages.
 
+config FLASH_HAS_DRIVER_ENABLED
```
Comment (galak) [inline arch/Kconfig]: This should live in drivers/flash/Kconfig.  Not sure why FLASH_HAS_PAGE_LAYOUT doesn't live there as well.

**Review by galak — CHANGES_REQUESTED** [review]
Looks good, we should move the one Kconfig symbol into drivers/flash/Kconfig

File: arch/Kconfig (hunk 1 above)
Comment (nvlsianpu) [reply arch/Kconfig]: If I had moved this to flash-Kconfig, the Flash driver options were placed paraller to FLASH menuconfig, not under it as I expected.

Comment (galak) [reply arch/Kconfig]: What do you mean parallel to it?

Comment (nvlsianpu) [reply arch/Kconfig]: image

Comment (nvlsianpu) [reply arch/Kconfig]: I found out that have to move the keword declaration before `menuconfig FLASH` for fix picture above.

Comment (galak) [reply arch/Kconfig]: that makes sense, we should do that.

Comment (nvlsianpu) [reply arch/Kconfig]: done

**Review by galak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
Thanks @nvlsianpu

**Pull request merged**